src/ResourceManager/AzureStackStorage/Commands.AzureStackStorage/Container/StartContainerMigration.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Management.Automation;
+using Microsoft.AzureStack.AzureConsistentStorage;
+using Microsoft.AzureStack.AzureConsistentStorage.Models;
+
+namespace Microsoft.AzureStack.AzureConsistentStorage.Commands
+{
+    using System.Collections;
+
+    /// <summary>
+    /// SYNTAX
+    /// Start-ACSContainerMigration  [-SubscriptionId] {string} [-Token] {string} [-AdminUri] {Uri} [-ResourceGroupName] {string} 
+    ///                  [-FarmName] {string} [-ShareName] {string} [-Container] {container} [-DestinationShareName] {string} [{CommonParameters}] 
+    /// </summary>
+    [Cmdlet(VerbsLifecycle.Start, Nouns.AdminContainerMigration)]
+    public sealed class StartContainerMigration : AdminCmdlet
+    {
+        /// <summary>
+        /// Resource group name
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 3)]
+        [ValidateNotNull]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Farm Identifier
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 4)]
+        [ValidateNotNull]
+        public string FarmName { get; set; }
+
+        /// <summary>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, Position = 5)]
+        [ValidateNotNullOrEmpty]
+        public Container ContainerToMigrate { get; set; }
```

Comment:
I meant how the user of this powershell would create the Container object to pass it as a parameter..